exampleplugins/5.2/SwagCustomStatistics/Controllers/Backend/SwagCustomStatistics.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+class Shopware_Controllers_Backend_SwagCustomStatistics extends Shopware_Controllers_Backend_ExtJs
+{
+    public function getVoucherStatisticsAction()
+    {
+        $additionalSelects = '';
+        $idList = (string) $this->Request()->getParam('selectedShops');
+
+        if (!empty($idList)) {
+            $selectedShopIds = explode(',', $idList);
+
+            foreach ($selectedShopIds as $shopId) {
+                $additionalSelects .= ', SUM(IF(vouchers.subshopID = ' . $shopId . ', codes.cashed, 0)) as amount' . $shopId . " \n";
```

Comment:
Possible SQL injection. If shopId is an int, just do a $shopId = (int) $shopId. Otherwise you could use the quote function from doctrine.

Reply:
done